### PR TITLE
 Fix crash Unread Widget Configuration

### PR DESCRIPTION
**Pull request by ndscor**
Please ensure that your pull request meets the following requirements - thanks !

* Follows our existing [codestyle](https://github.com/k9mail/k-9/wiki/CodeStyle).
* Does not break any unit tests.
* Contains a reference to the issue that it fixes.
* For cosmetic changes add one or multiple images, if possible.

Prevent crash when trying to choose folders without selecting a account.

### CODE REVIEW

**Comment by cketti** [comment]
@nitmee: Can you please add some context to this pull request. What problem are you trying to solve? Can you provide instructions on how to reproduce the problem?

**Comment by nitmee** [comment]
the crash occurred when trying to configure unread widget . go to widgets select K-9 Mail / Configure K-9 Unread  .
Screenshot_20200602-172655_K-9 Mail

Without selecting accounts , Check folder count and tap to select folders , K-9 Mail will crash because no account selected .

**Comment by cketti** [comment]
The folder-related settings should be disabled until an account is selected. It appears you're trying to fix a symptom rather than the cause.

File: app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt [hunk 1]
```diff
@@ -46,18 +46,23 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
         }
 
         unreadFolderEnabled = findPreference(PREFERENCE_UNREAD_FOLDER_ENABLED)!!
-        unreadFolderEnabled.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, _ ->
-            unreadFolder.summary = getString(R.string.unread_widget_folder_summary)
-            selectedFolderId = null
-            true
+        if (selectedAccountUuid == null) {
```
Comment (cketti) [inline app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt:49]: At the point where the fragment is created `selectedAccountUuid` is always null. So there's no need for this check.

File: app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt [hunk 2]
```diff
@@ -46,18 +46,23 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
...
+            unreadFolderEnabled.isEnabled = false
+            unreadFolderEnabled.isChecked = false
+        } else {
```
Comment (cketti) [inline app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt:52]: This case is never executed. But it should. The listener is responsible for resetting the "Folder" preference description to the default when the "Folder count" preference is unchecked (and a folder was selected before).

Steps to test is works as intended (like it did before your change):
1. Select an account
2. Check "Folder count"
3. Tap "Folder" preference and select a folder -> Folder name is displayed as preference "description"
4. Uncheck "Folder count" -> Folder name "description" should be reset to default and no longer show the folder name

File: app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt [hunk 3]
```diff
@@ -46,18 +46,23 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
...
         unreadFolder = findPreference(PREFERENCE_UNREAD_FOLDER)!!
-        unreadFolder.onPreferenceClickListener = Preference.OnPreferenceClickListener {
+     unreadFolder.onPreferenceClickListener = Preference.OnPreferenceClickListener {
             val intent = ChooseFolderActivity.buildLaunchIntent(
-                context = requireContext(),
-                accountUuid = selectedAccountUuid!!,
-                showDisplayableOnly = true
+                    context = requireContext(),
+                    accountUuid = selectedAccountUuid!!,
+                    showDisplayableOnly = true
```
Comment (cketti) [inline app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt:65]: All of these changes are unnecessary. Please remove them.

File: app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt [hunk 4]
```diff
@@ -46,18 +46,23 @@ class UnreadWidgetConfigurationFragment : PreferenceFragmentCompat() {
         }
 
         unreadFolderEnabled = findPreference(PREFERENCE_UNREAD_FOLDER_ENABLED)!!
-        unreadFolderEnabled.onPreferenceChangeListener = Preference.OnPreferenceChangeListener { _, _ ->
-            unreadFolder.summary = getString(R.string.unread_widget_folder_summary)
-            selectedFolderId = null
-            true
+        if (selectedAccountUuid == null) {
+            unreadFolderEnabled.isEnabled = false
+            unreadFolderEnabled.isChecked = false
```
Comment (cketti) [inline app/k9mail/src/main/java/com/fsck/k9/widget/unread/UnreadWidgetConfigurationFragment.kt:51]: This line isn't really necessary. The checkbox is unchecked by default.

**Comment by ndscor** [comment]
> Thanks for contributing to K-9 Mail 👍
> 
> This pull request needs a little bit more work. I've added a few comments.
> 
> A few tips for the future:
> 
> * Don't create pull requests from your fork's `master` branch. Typically that's the one you keep in sync with the `master` branch of the upstream repository. Create a new branch for each pull request.
> * The status of this pull request is red. "Some checks were not successful". You can click through to the build log to find out what was causing the failure. In this case it's the fact that you introduced code style violations (wrong indentation levels). These issues can be fixed automatically. See https://github.com/k9mail/k-9/wiki/BuildingK9#checkfix-code-style
> * We try to keep our Git history as tidy as possible. Ideally this fix would consist of exactly one commit. Please feel free to force-push to your branch to update the PR.
> 
> I believe if you apply all my suggestions this will become a one line change.


@cketti thanks and appreciate your suggestion , I have made changes and yes it become one line change. do you want me to delete this request and create a new one with a new branch ?

**Comment by cketti** [comment]
> do you want me to delete this request and create a new one with a new branch ?

I think that would be easier than trying to clean up the Git history of this branch.

**Comment by ndscor** [comment]
@cketti new pull request #4827

**Pull request closed without merging**